core/codegen/src/decorators/derive_form.rs

```diff
@@ -27,11 +27,11 @@ fn struct_lifetime(ecx: &mut ExtCtxt, item: &Annotatable, sp: Span) -> Option<St
             ItemKind::Struct(_, ref generics) => {
                 let mut lifetimes = generics.params.iter()
                     .filter_map(|p| match *p {
-                        GenericParam::Lifetime(ref def) => Some(def),
+                        GenericParam { ref ident, ref kind, .. } if *kind == GenericParamKind::Lifetime => Some(ident),
```

Comment:
This is the only part I'm not quite certain of. Seems to work, but there might be a better way of doing this.

Reply:
Does `ident` include the initial tick `'`? You should take a look at the generated code and ensure that it looks correct, even if it's accepted by the compiler otherwise. Aside from this, this looks good, though it would seem that `kind` doesn't need to be a `ref kind`.

Reply:
I'm pushing a slightly different fix for this in 0.3. Let's use the same one for master.